### PR TITLE
Governance update

### PR DESCRIPTION
**Pull request by siovanus**
fix some overflow bugs

add change view block num constrain

fix if consensus bug

### CODE REVIEW

File: smartcontract/service/native/governance/method.go [hunk 1]
```diff
@@ -913,13 +915,16 @@ func executeSplit2(native *native.NativeService, contract common.Address, view u
 
 	//fee split of consensus peer
 	for i := 0; i < int(config.K); i++ {
-		nodeAmount := nodeIncome * uint64(globalParam.A) / 100 * peersCandidate[i].S / sumS
-		err = splitNodeFee(native, contract, peersCandidate[i].PeerPubkey, peersCandidate[i].Address, true,
-			peerPoolMap.PeerPoolMap[peersCandidate[i].PeerPubkey].TotalPos, nodeAmount)
+		nodeAmount := new(big.Int).Div(new(big.Int).Mul(new(big.Int).Div(new(big.Int).Mul(nodeIncome,
+			new(big.Int).SetUint64(uint64(globalParam.A))), new(big.Int).SetUint64(100)),
```
Comment (laizy) [inline smartcontract/service/native/governance/method.go:923]: can this be simplified? it is hard to read.

File: smartcontract/service/native/governance/utils.go [hunk 2]
```diff
@@ -204,13 +204,14 @@ func splitCurve(native *native.NativeService, contract common.Address, pos uint6
...
 		index = uint64(len(Xi) - 2)
+		xi = uint64(Xi[len(Xi)-1])
 	}
 	splitCurve, err := getSplitCurve(native, contract)
 	if err != nil {
 		return 0, fmt.Errorf("getSplitCurve, get splitCurve error: %v", err)
 	}
 	Yi := splitCurve.Yi
-	s := ((uint64(Yi[index+1])-uint64(Yi[index]))*xi + uint64(Yi[index])*uint64(Xi[index+1]) - uint64(Yi[index+1])*uint64(Xi[index])) / (uint64(Xi[index+1]) - uint64(Xi[index]))
+	s := (uint64(Yi[index+1])*xi - uint64(Yi[index])*xi + uint64(Yi[index])*uint64(Xi[index+1]) - uint64(Yi[index+1])*uint64(Xi[index])) / (uint64(Xi[index+1]) - uint64(Xi[index]))
```
Comment (Honglei-Cong) [inline smartcontract/service/native/governance/utils.go]: use int64, then abs(int64).

File: smartcontract/service/native/governance/method.go (hunk 1 above)
Comment (Honglei-Cong) [reply smartcontract/service/native/governance/method.go:923]: yes. 
Or left the old code as comment, to help reading.

File: smartcontract/service/native/governance/utils.go (hunk 2 above)
Comment (siovanus) [reply smartcontract/service/native/governance/utils.go]: finalize as this

**Review by laizy — APPROVED** [review]

**Pull request merged**